web/src/components/LandingPageSlider.tsx

```diff
@@ -0,0 +1,176 @@
+import { LandingPageActionTile } from "@/components/LandingPageActionTile";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { QUERIES, ROUTES, routeWithQuery } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { useRouter } from "next/router";
+import { ReactElement, useRef, useState } from "react";
+import Slider from "react-slick";
+import "slick-carousel/slick/slick-theme.css";
+import "slick-carousel/slick/slick.css";
+import { useMountEffect } from "../lib/utils/helpers";
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+function SampleNextArrow(props: any) {
+  const { onClick } = props;
+  return (
+    <button onClick={onClick} className="slider-button next">
+      {">"}
+    </button>
+  );
+}
+function SamplePrevArrow(props: any) {
+  const { onClick } = props;
+  return (
+    <button onClick={onClick} className="slider-button prev">
+      {"<"}
+    </button>
+  );
+}
+
+export const LandingPageSlider = (): ReactElement => {
+  const router = useRouter();
+  const { Config } = useConfig();
+  const [activeSlide, setActiveSlider] = useState(0);
+  const [slideBreakPoint, setSlideBreakPoint] = useState(7);
+  const cardRef = useRef<null | HTMLButtonElement>(null);
+  const sliderRef = useRef<null | HTMLDivElement>(null);
+
+  const getNumberOfTilesToShow = (): number => {
+    const sliderWidth = sliderRef.current?.clientWidth;
+    const cardWidth = cardRef.current?.parentElement?.clientWidth;
+    if (!sliderWidth || !cardWidth) return 7;
+    const numberOfTilesToFit = Math.ceil(sliderWidth / (cardWidth + 100));
+    return 7 - numberOfTilesToFit;
+  };
+
+  useMountEffect(() => {
+    setSlideBreakPoint(getNumberOfTilesToShow());
+    function handleResize() {
+      setSlideBreakPoint(getNumberOfTilesToShow());
+    }
+    window.addEventListener("resize", handleResize);
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  });
+
+  const routeToOnboarding = () => {
+    router.push(ROUTES.onboarding);
+    analytics.event.landing_page_hero_get_started.click.go_to_onboarding();
+  };
+
+  const setFlowAndRouteUser = (flow: "starting" | "out-of-state" | "up-and-running") => {
+    routeWithQuery(router, {
+      path: ROUTES.onboarding,
+      queries: { [QUERIES.flow]: flow },
+    });
+  };
+
+  const settings = {
+    className: "slider variable-width",
+    slidesToShow: 1,
```

Comment:
Can you explain this to me? It's working, but how is it working when the slidesToShow is always set to 1?